2020_10_02-traefik-load-balancing/Makefile

```diff
@@ -1,93 +1,115 @@
-LOCAL_BIN=~/.local/bin/
-PACKER_VERSION=1.6.2
-TERRAFORM_PLUGIN_DIR=~/.terraform.d/plugins
-TERRAFORM_VERSION=0.13.3
-TERRAFORM_LIBVIRT_VERSION=0.6.2
-TERRAFORM_ANSIBLE_VERSION=2.3.3
-ANSIBLE_VERSION=2.9.13
-LIBVIRT_HYPERVISOR_URI="qemu:///system"
-LIBVIRT_IMAGES_POOL="templates"
-LIBVIRT_IMAGE_NAME="debian10-traefik.qcow2"
-ROOT_PASSWORD="traefik"
+.PHONY: help all check-env my-linux-env create-env install-packer install-terraform install-ansible install-terraform-plugins image build-image upload-image import-kube-nodes create-vms run-playbook destroy
+.DEFAULT_GOAL := all
+
+LOCAL_BIN := ~/.local/bin/
+PACKER_VERSION := 1.6.5
+TERRAFORM_PLUGIN_DIR := ~/.terraform.d/plugins
+TERRAFORM_VERSION := 0.13.5
+TERRAFORM_LIBVIRT_FULL_VERSION := 0.6.3+git.1604843676.67f4f2aa.Ubuntu_20.04.amd64
+TERRAFORM_LIBVIRT_VERSION := 0.6.3 #if you change that, change also terraform/traefik.tf
+TERRAFORM_ANSIBLE_VERSION := 2.4.0
+ANSIBLE_VERSION := 2.9.13
+LIBVIRT_HYPERVISOR_URI := "qemu:///system"
+LIBVIRT_TEMPLATES_IMAGES_POOL := "templates"
+LIBVIRT_TEMPLATES_IMAGES_POOL_DIR := "/var/lib/libvirt/images/templates"
+LIBVIRT_IMAGE_NAME := "debian10-traefik.qcow2"
+ROOT_PASSWORD := "traefik"
+CLUSTER := 1
+TRAEFIKEE_LICENSE := "N/A"
 $(eval SSH_IDENTITY=$(shell find ~/.ssh/ -name 'id_*' -not -name '*.pub' | head -n 1))
-CLUSTER=1
-TRAEFIKEE_LICENSE="N/A"
+#SSH_IDENTITY := "~/.ssh/id_rsa_unencrypted"
+
+help: ## This help.
+	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-]+:.*?## / {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
 
-all:
+all: my-linux-env image run-playbook ## Do everything.
```

Comment:
If you really want to do everything (which I didn't choose) just typing make, then check-env is missing.

Reply:
more over make should check you're on linux because just "make" won't work on macOS for instance